alacritty/src/event.rs

```diff
@@ -66,6 +65,374 @@ const MAX_SEARCH_HISTORY_SIZE: usize = 255;
+    pub fn create_initial_window(
+        &mut self,
+        event_loop: &ActiveEventLoop,
+        options: WindowOptions,
+    ) -> Result<(), Box<dyn Error>> {
+        let window_context =
+            WindowContext::initial(event_loop, self.proxy.clone(), self.config.clone(), options)?;
+
+        self.gl_display = Some(window_context.display.gl_context().display());
+        self.windows.insert(window_context.id(), window_context);
+
+        Ok(())
+    }
+
+    /// Create a new terminal window.
+    pub fn create_window(
+        &mut self,
+        event_loop: &ActiveEventLoop,
+        options: WindowOptions,
+    ) -> Result<(), Box<dyn Error>> {
+        let window = self.windows.iter().next().as_ref().unwrap().1;
+
+        // Overide config with CLI/IPC options.
+        let mut config_overrides = options.config_overrides();
+        #[cfg(unix)]
+        config_overrides.extend_from_slice(&self.global_ipc_options);
+        let mut config = self.config.clone();
+        config = config_overrides.override_config_rc(config);
+
+        #[allow(unused_mut)]
+        let mut window_context =
+            window.additional(event_loop, self.proxy.clone(), config, options, config_overrides)?;
+
+        self.windows.insert(window_context.id(), window_context);
+        Ok(())
+    }
+
+    /// Run the event loop.
+    ///
+    /// The result is exit code generate from the loop.
+    pub fn run(mut self, event_loop: EventLoop<Event>) -> Result<(), Box<dyn Error>> {
+        let result = event_loop.run_app(&mut self);
+        if let Some(initial_window_error) = self.initial_window_error.take() {
+            Err(initial_window_error)
+        } else {
+            result.map_err(Into::into)
+        }
+    }
+
+    /// Check if an event is irrelevant and can be skipped.
+    fn skip_window_event(event: &WindowEvent) -> bool {
+        matches!(
+            event,
+            WindowEvent::KeyboardInput { is_synthetic: true, .. }
+                | WindowEvent::ActivationTokenDone { .. }
+                | WindowEvent::DoubleTapGesture { .. }
+                | WindowEvent::TouchpadPressure { .. }
+                | WindowEvent::RotationGesture { .. }
+                | WindowEvent::CursorEntered { .. }
+                | WindowEvent::PinchGesture { .. }
+                | WindowEvent::AxisMotion { .. }
+                | WindowEvent::PanGesture { .. }
+                | WindowEvent::HoveredFileCancelled
+                | WindowEvent::Destroyed
+                | WindowEvent::ThemeChanged(_)
+                | WindowEvent::HoveredFile(_)
+                | WindowEvent::Moved(_)
+        )
+    }
+}
+
+impl ApplicationHandler<Event> for Processor {
+    fn resumed(&mut self, _event_loop: &ActiveEventLoop) {
+        // We use `new_events` because we target only desktop.
+    }
+
+    fn new_events(&mut self, event_loop: &ActiveEventLoop, cause: StartCause) {
+        if cause == StartCause::Init {
```

Comment:
Early return for a lil less indentation?

Reply:
Last time the style for thing like that was the one I picked.

I can do early return, but I don't think in this particular case it's really needed, because indention is low.

Reply:
I mean it's 4 levels of indentation already. With kernel indentation rules that would be 32 columns of indentation alone for the deepest level. I wouldn't call that "low".

If there were plans to handle different causes in the future, it might be different. But I don't see any immediate reason why we **ever** would handle different causes, and if we did, we'd probably switch to a match anyway. So I fail to see how an early return isn't the best option here.